internal/driver/driver.go

```diff
@@ -562,10 +596,40 @@ func (d *Driver) getPathName(device *Device, queryParams url.Values) (string, er
 				fmt.Sprintf("Video streaming path does not exist for the device %v at PathIndex %d", device.name, pathIndexConv), nil)
 		}
 		videoPath = device.paths[pathIndexConv]
+	} else if streamFormat != "" { // use stream format video path value
+		if streamFormat != RGB && streamFormat != Greyscale && streamFormat != Depth {
+			return "", errors.NewCommonEdgeX(errors.KindIOError, "Invalid stream format. Valid options are 'RGB', 'Greyscale', or 'Depth.'", nil)
+		}
+		for _, path := range device.paths {
+			videoPath, err := getStreamFormatPath(path, streamFormat)
+			if err != nil {
+				continue
+			}
+			return videoPath, nil
+		}
+		return "", errors.NewCommonEdgeX(errors.KindIOError, fmt.Sprintf("Invalid stream format for device %s.", device.name), nil)
 	}
 	return videoPath, nil
 }
 
+func getStreamFormatPath(path string, streamFormat string) (string, error) {
```

Comment:
1. Path does not need to be returned, as you are just returning the input argument
2. The name of the function should be more clear, as this is more of a comparison/match than a get
3. I would recommend changing it so that the function returns just a bool whether or not it matches
4. I would make it be a driver receiver to be able to log the errors


```go
func (d *Driver) pathMatchesStreamFormat(path, streamFormat string) bool {
    ...
    if err != nil {
      d.lc.Errorf("blah")
      return false 
   }
...
return  currentType == streamFormat

}
```

And then up above inside of `getPathName`, do something like so:
```go
		for _, path := range device.paths {
			if d.pathMatchesStreamFormat(path, streamFormat) {
                            return path
                        }
		}
```